components/list/list-header.js

```diff
@@ -66,14 +75,32 @@ class ListHeader extends RtlMixin(LocalizeCoreElement(LitElement)) {
 			:host([hidden]) {
 				display: none;
 			}
-			.d2l-list-header-container {
+			:host([requires-selection]) .d2l-list-header-container {
+				display: none;
+				max-height: 0;
+				opacity: 0;
+				/* transform: translate(0, -10px); */
+				transition: opacity 100ms linear, max-height 100ms linear;
+			}
+			:host([requires-selection]) .d2l-list-header-container-pre-showing,
+			:host([requires-selection]) .d2l-list-header-container-hiding {
+				display: block;
+			}
+			:host([requires-selection]) .d2l-list-header-container-showing {
+				display: block;
+				max-height: 70px;
```

Comment:
This assumes a `max-height`, which is probably not a safe assumption since consumers can potentially put all sorts of things into the header. This assumption is probably worth discussing with Design, but ultimately, a height measurement may be more appropriate if we are required to animate the height aspect.

Reply:
What are you picturing for a height measurement? I'll probably need to measure this component's height to get sticky table headers working too.

Reply:
In this case, if we needed the height of the contents, in `updated()` I would probably try to use `getBoundingClientRect()` on the inner `div` after `display: block` is applied to the outer `div` but before beginning the animation. 

While the outer `div` would have a `height` of 0 (because `max-height: 0`), the inner `div` would be able to provide the height of the contents. I would then assign it to a  `_contentHeight` property, which could be used in `render()` to specify either `height` or `max-height` style property on the outer `div`, and the `transition` would kick in at that point.

If we measure before beginning each `showing` animation, then we don't need to worry if the consumer adds/removes something from the header that affects the height. We'd never expect the contents of the header's slot to be changed during the animation. The explicit height assignment would be removed after the animation is complete.

So those are my initial thoughts.